### ISSUE TITLE
Nested reroute doesnt accept input

### ISSUE THREAD

**Issue by Lalimec** [ssitu/ComfyUI_NestedNodeBuilder#2]
I am trying to use reroute to reduce the amount of connections to the nested node. however after nesting the group i cant run it. 

Here is the full pipeline. Result is correct.
![image](https://github.com/ssitu/ComfyUI_NestedNodeBuilder/assets/65521426/8c345312-6f52-40e6-bc3a-9edc989f68df)

After nesting the nodes they stop working. As you can see input images are not used.
![image](https://github.com/ssitu/ComfyUI_NestedNodeBuilder/assets/65521426/4310eda8-61e2-49e7-b97e-931959923ecf)

Unnesting the node shows the missing inputs. (Images get disconnected automatically after unnesting, it is not done manually)
![image](https://github.com/ssitu/ComfyUI_NestedNodeBuilder/assets/65521426/521e01b9-ea37-4c06-af60-2f6005449648)

I am not sure what the problem is, but it looks like there is a weird behavior while using reroute. Removing them solves the issue.
![image](https://github.com/ssitu/ComfyUI_NestedNodeBuilder/assets/65521426/e88f96b6-5098-497f-995e-118fd0e8da5f)

```
ubuntu@ip-172-31-44-50:~/ComfyUI$ ./comfy.sh
** ComfyUI start up time: 2023-08-10 22:15:11.112237

Prestartup times for custom nodes:
   0.0 seconds: /home/ubuntu/ComfyUI/custom_nodes/ComfyUI-Manager

Total VRAM 22516 MB, total RAM 31751 MB
xformers version: 0.0.20
Set vram state to: NORMAL_VRAM
Device: cuda:0 NVIDIA A10G : cudaMallocAsync
Using xformers cross attention
Adding extra search path checkpoints /home/ubuntu/vlad/models/Stable-diffusion
Adding extra search path configs /home/ubuntu/vlad/models/Stable-diffusion
Adding extra search path vae /home/ubuntu/vlad/models/VAE
Adding extra search path loras /home/ubuntu/vlad/models/Lora
Adding extra search path loras /home/ubuntu/vlad/models/LyCORIS
Adding extra search path upscale_models /home/ubuntu/vlad/models/ESRGAN
Adding extra search path upscale_models /home/ubuntu/vlad/models/RealESRGAN
Adding extra search path upscale_models /home/ubuntu/vlad/models/SwinIR
Adding extra search path embeddings /home/ubuntu/vlad/embeddings
Adding extra search path hypernetworks /home/ubuntu/vlad/models/hypernetworks
Adding extra search path controlnet /home/ubuntu/vlad/extensions-builtin/sd-webui-controlnet/models
### Loading: ComfyUI-Manager (V0.21)
### ComfyUI Revision: 1271 [f7e6a5ed]
### Loading: ComfyUI-Impact-Pack (V3.12.1)
### Loading: ComfyUI-Impact-Pack (Subpack: V0.2)
Total VRAM 22516 MB, total RAM 31751 MB
xformers version: 0.0.20
Set vram state to: NORMAL_VRAM
Device: cuda:0 NVIDIA A10G : cudaMallocAsync
WAS Node Suite: BlenderNeko's Advanced CLIP Text Encode found, attempting to enable `CLIPTextEncode` support.
WAS Node Suite: `CLIPTextEncode (BlenderNeko Advanced + NSP)` node enabled under `WAS Suite/Conditioning` menu.
WAS Node Suite: OpenCV Python FFMPEG support is enabled
WAS Node Suite Warning: `ffmpeg_bin_path` is not set in `/home/ubuntu/ComfyUI/custom_nodes/was-node-suite-comfyui/was_suite_config.json` config file. Will attempt to use system ffmpeg binaries if available.
WAS Node Suite: Finished. Loaded 191 nodes successfully.

        "Don't be afraid to give up the good to go for the great." - John D. Rockefeller

/home/ubuntu/ComfyUI/custom_nodes/failfast-comfyui-extensions/extensions
/home/ubuntu/ComfyUI/web/extensions/failfast-comfyui-extensions
Using xformers cross attention

Import times for custom nodes:
   0.0 seconds: /home/ubuntu/ComfyUI/custom_nodes/bsz-auto-hires.py
   0.0 seconds: /home/ubuntu/ComfyUI/custom_nodes/ComfyUI_Noise
   0.0 seconds: /home/ubuntu/ComfyUI/custom_nodes/ComfyUI_ADV_CLIP_emb
   0.0 seconds: /home/ubuntu/ComfyUI/custom_nodes/ComfyUI_Cutoff
   0.0 seconds: /home/ubuntu/ComfyUI/custom_nodes/failfast-comfyui-extensions
   0.0 seconds: /home/ubuntu/ComfyUI/custom_nodes/images-grid-comfy-plugin
   0.0 seconds: /home/ubuntu/ComfyUI/custom_nodes/ComfyMath
   0.0 seconds: /home/ubuntu/ComfyUI/custom_nodes/ComfyUI_NestedNodeBuilder
   0.0 seconds: /home/ubuntu/ComfyUI/custom_nodes/ComfyUI_UltimateSDUpscale
   0.0 seconds: /home/ubuntu/ComfyUI/custom_nodes/Derfuu_ComfyUI_ModdedNodes
   0.0 seconds: /home/ubuntu/ComfyUI/custom_nodes/ComfyUI-Manager
   0.1 seconds: /home/ubuntu/ComfyUI/custom_nodes/ComfyUI_smZNodes
   0.1 seconds: /home/ubuntu/ComfyUI/custom_nodes/comfyui-dynamicprompts
   0.2 seconds: /home/ubuntu/ComfyUI/custom_nodes/text-test
   0.4 seconds: /home/ubuntu/ComfyUI/custom_nodes/comfy_controlnet_preprocessors
   0.4 seconds: /home/ubuntu/ComfyUI/custom_nodes/efficiency-nodes-comfyui
   0.8 seconds: /home/ubuntu/ComfyUI/custom_nodes/ComfyUI-Impact-Pack
   0.9 seconds: /home/ubuntu/ComfyUI/custom_nodes/ComfyUI_Inpaint
   1.0 seconds: /home/ubuntu/ComfyUI/custom_nodes/was-node-suite-comfyui

Starting server

To see the GUI go to: http://0.0.0.0:8188
```

**Comment by ssitu** [comment]
Could you provide the exact workflow? I can't seem to recreate the issue on mine: 

https://github.com/ssitu/ComfyUI_NestedNodeBuilder/assets/57548627/cdeb2174-81df-4ca3-b0ca-227961d4f166

**Comment by Lalimec** [comment]
well this is weird. Maybe some of the nodes im using interfere with nested nodes. nodes are in the sample img.

![image](https://github.com/ssitu/ComfyUI_NestedNodeBuilder/assets/65521426/fb72e849-92d6-478c-ae73-50b0c66d5a53)

 
![ComfyUI_temp_paqlp_00002_](https://github.com/ssitu/ComfyUI_NestedNodeBuilder/assets/65521426/f51466c5-2e9e-473d-8f12-29b6865f5ab6)

**Comment by ssitu** [comment]
It looks like your reroutes are different than my reroute nodes. The one with the arrow is the one that I created. 
![image](https://github.com/ssitu/ComfyUI_NestedNodeBuilder/assets/57548627/55d44e24-dd83-4075-9c2c-eded9e4c4490)

Here is the same workflow but with my reroutes. See if this works for you. Uploaded txt because .json isn't supported:

[issue2_fixed_reroutes.txt](https://github.com/ssitu/ComfyUI_NestedNodeBuilder/files/12327672/issue2_fixed_reroutes.txt)

**Comment by Lalimec** [comment]
How is that possible lol? I thought those are default nodes that come with comfy.

**Comment by ssitu** [comment]
For the infinite unnest problem, I just added some code to handle that in the linked pull request that will fix it. There's also an unconnected reroute node hiding under the second reroute which was causing the issue. I will try to make a fix the first workflow momentarily.

**Comment by Lalimec** [comment]
Great, will let you know if there are any other cases like these.

**Comment by ssitu** [comment]
I made some changes to fix the two recent workflows that you gave me. Are you able to try https://github.com/ssitu/ComfyUI_NestedNodeBuilder/pull/5 to see if it fixes your problems so far?

**Comment by Lalimec** [comment]
works like a charm! i was wondering if there is a way to nest reroutes only? I tried to include a pseudo node in the test but realized that the order of the outputs gets all mixed up.

![image](https://github.com/ssitu/ComfyUI_NestedNodeBuilder/assets/65521426/f21a67e3-eb17-41d1-9f8d-6667e4013d08)
![image](https://github.com/ssitu/ComfyUI_NestedNodeBuilder/assets/65521426/ba96da03-f7bd-4a5b-aead-93bd2907b0ad)

**Comment by ssitu** [comment]
> Does it order according to the node id or sth?
Yeah, the order of the inputs/outputs are based on the internal node id, so it will be according to the order of when the nodes were created. I could change it to something else but I'm not sure what would be a good way to order them.

I will go see if I can fix problems with consecutive reroutes.

**Comment by ssitu** [comment]
Alright I pushed a fix for those cases to #5, see if that fixes the problem for you.

**Comment by Lalimec** [comment]
> > Does it order according to the node id or sth?
> 
> Yeah, the order of the inputs/outputs are based on the internal node id, so it will be according to the order of when the nodes were created. I could change it to something else but I'm not sure what would be a good way to order them.
> 
> I will go see if I can fix problems with consecutive reroutes.

Maybe the sum of x y coordinates in ascending order could be helpful but I am not sure. Thank you for your work btw, it is an amazing feature really.

Here is the test network. 
![image](https://github.com/ssitu/ComfyUI_NestedNodeBuilder/assets/65521426/f98218d9-e065-4dae-bb7a-29fd3c71ba4d)

Nested result. Floating reroute creates both input and output but other than that it seems to work great!
![image](https://github.com/ssitu/ComfyUI_NestedNodeBuilder/assets/65521426/a26f331b-83c2-43ae-a1ba-9ae6bc9aa0ec)

And just out of curiosity i ve attached image node to the surplus socket... And weirdly enough it worked like it should lol. 😆 
![image](https://github.com/ssitu/ComfyUI_NestedNodeBuilder/assets/65521426/4425b255-9793-4408-99b1-ea6ca3a855dd)
![image](https://github.com/ssitu/ComfyUI_NestedNodeBuilder/assets/65521426/25d7fcaf-2716-478b-8702-810891ca2a08)

The only problem is when you drag a connection from any of the sockets to create a node, new node doesnt connect to it. Insted it connects to the top one for some reason. connecting an existing node works as it should though. And it is not related to reroute nodes.
![image](https://github.com/ssitu/ComfyUI_NestedNodeBuilder/assets/65521426/4351537d-8527-4f57-9dc3-9f85d2819316)
![image](https://github.com/ssitu/ComfyUI_NestedNodeBuilder/assets/65521426/579d569b-4cba-430d-9d37-94b13716f375)

![image](https://github.com/ssitu/ComfyUI_NestedNodeBuilder/assets/65521426/5abd90ba-55ce-4a35-94b8-4f8da455691c)
![image](https://github.com/ssitu/ComfyUI_NestedNodeBuilder/assets/65521426/1e830a89-40cb-4027-9ab2-099dec29b3ae)

**Comment by ssitu** [comment]
The wrong slot connection problem is just a problem with Litegraph. You get the same problem with other custom nodes that have outputs with the same name, like the Image Padding node from WAS Suite.

![image](https://github.com/ssitu/ComfyUI_NestedNodeBuilder/assets/57548627/06450b0b-f152-481c-bf9e-b039c3dde75e)
![image](https://github.com/ssitu/ComfyUI_NestedNodeBuilder/assets/57548627/a133458b-ffa4-4482-948c-1f9da6e30385)

**Comment by ssitu** [comment]
Merged the current fixes so that others can use them. Open a new issue if you find any other problems. I really appreciate you testing this, it's rare for people to open issues so I'm not aware of all the issues.

**Comment by Lalimec** [comment]
No problem at all. I use sd in my work a lot, so getting things fixed is good for me too haha. Thans for the attention, much appreciated.

**Comment by ssitu** [comment]
I tried installing all the required extensions for the workflow, but the manager doesn't recognize the TextOnCanvas node. Where can I install that node?

**Comment by ssitu** [comment]
Yeah, the reroutes are being disconnected for me too, I'll look into it.

**Comment by Lalimec** [comment]
Tex on cancas is the node  wrote, shouldn't be needing that. But i can send if tou want haha.

**Comment by ssitu** [comment]
Sorry about that, I misread what the problematic grouping was. 

Anyway, I made some more changes in #7 and got the last grouping to nest and unnest as intended. There isn't many nodes in the base install with optional inputs, so I mistakenly never tested how they would interact with this addon, so I had a tiny bug in the code for that. 
If you don't mind, see if the changes made in #7 doesn't affect your other nested nodes when you nest/unnest/execute with them.

**Comment by ssitu** [comment]
https://github.com/ssitu/ComfyUI_NestedNodeBuilder/assets/57548627/ac74e89f-d369-4424-8396-bf1dee076711

It's working for me, are you using the changes from #7?

**Comment by ssitu** [comment]
I'm going to merge these changes. I've done testing on my end and these changes seem to be pretty important for making this extension less buggy.

**Comment by Lalimec** [comment]
sorry for the late respond, i didnt had time to ckeck it out. but my comy installation is somehow broken i guess. After switching to the main branch, since you merged it, i cant unnest the node. i only get 2 blue nodes outside and the rest stays nested. I have bunch of custom nodes installed, are you aware of any known conflicts.

![image](https://github.com/ssitu/ComfyUI_NestedNodeBuilder/assets/65521426/4e37b586-5537-43d7-b189-ad7b3b47e4ed)

```
(comfy) ubuntu@ip-172-31-44-50:~/ComfyUI$ bash comfy.sh
** ComfyUI start up time: 2023-08-25 10:34:29.199321

Prestartup times for custom nodes:
   0.0 seconds: /home/ubuntu/ComfyUI/custom_nodes/ComfyUI-Manager

Total VRAM 22516 MB, total RAM 31751 MB
xformers version: 0.0.20
2023-08-25 10:34:31.592014: I tensorflow/core/platform/cpu_feature_guard.cc:182] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.
To enable the following instructions: AVX2 FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.
2023-08-25 10:34:32.257972: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Could not find TensorRT
Set vram state to: NORMAL_VRAM
Device: cuda:0 NVIDIA A10G : cudaMallocAsync
Using xformers cross attention
Adding extra search path checkpoints /home/ubuntu/vlad/models/Stable-diffusion
Adding extra search path configs /home/ubuntu/vlad/models/Stable-diffusion
Adding extra search path vae /home/ubuntu/vlad/models/vae
Adding extra search path loras /home/ubuntu/vlad/models/Lora
Adding extra search path loras /home/ubuntu/vlad/models/LyCORIS
Adding extra search path upscale_models /home/ubuntu/vlad/models/ESRGAN
Adding extra search path upscale_models /home/ubuntu/vlad/models/RealESRGAN
Adding extra search path upscale_models /home/ubuntu/vlad/models/SwinIR
Adding extra search path embeddings /home/ubuntu/vlad/models/embeddings
Adding extra search path hypernetworks /home/ubuntu/vlad/models/hypernetworks
Adding extra search path controlnet /home/ubuntu/vlad/extensions-builtin/sd-webui-controlnet/models
/opt/conda/envs/comfy/lib/python3.10/site-packages/torchvision/transforms/functional_tensor.py:5: UserWarning: The torchvision.transforms.functional_tensor module is deprecated in 0.15 and will be **removed in 0.17**. Please don't rely on it. You probably just need to use APIs in torchvision.transforms.functional or in torchvision.transforms.v2.functional.
  warnings.warn(
[comfy_mtb] | INFO -> Loaded the following nodes:
        Face Swap (mtb): Face swap using deepinsight/insightface models
        Load Face Swap Model (mtb): Loads a faceswap model
        Load Face Analysis Model (mtb): Loads a face analysis model
        Restore Face (mtb): Uses GFPGan to restore faces
        Load Face Enhance Model (mtb): Loads a GFPGan or RestoreFormer model for face enhancement.
        Load Film Model (mtb): Loads a FILM model
        Film Interpolation (mtb): Google Research FILM frame interpolation for large motion
        Concat Images (mtb): Add images to batch
        Color Correct (mtb): Various color correction methods
        Image Compare (mtb): Compare two images and return a difference image
        Blur (mtb): Blur an image using a Gaussian filter.
        Mask To Image (mtb): Converts a mask (alpha) to an RGB image with a color and background
        Colored Image (mtb): Constant color image of given size
        Image Premultiply (mtb): Premultiply image with mask
        Image Resize Factor (mtb): Extracted mostly from WAS Node Suite, with a few edits (most notably multiple image support) and less features.
        Save Image Grid (mtb): Save all the images in the input batch as a grid of images.
        Load Image From Url (mtb): Load an image from the given URL
        Debug (mtb): Experimental node to debug any Comfy values, support for more types and widgets is planned
        Save Tensors (mtb): Save torch tensors (image, mask or latent) to disk, useful to debug things outside comfy
        Animation Builder (mtb): Convenient way to manage basic animation maths at the core of many of my workflows
        Float To Number (mtb): Node addon for the WAS Suite. Converts a "comfy" FLOAT to a NUMBER.
        Int To Bool (mtb): Basic int to bool conversion
        Int To Number (mtb): Node addon for the WAS Suite. Converts a "comfy" INT to a NUMBER.
        Smart Step (mtb): Utils to control the steps start/stop of the KAdvancedSampler in percentage
        Styles Loader (mtb): Load csv files and populate a dropdown from the rows (à la A111)
        Interpolate Clip Sequential (mtb): -
        Load Image Sequence (mtb): Load an image sequence from a folder. The current frame is used to determine which image to load.
        Save Image Sequence (mtb): Save an image sequence to a folder. The current frame is used to determine which image to save.
        Deep Bump (mtb): Normal & height maps generation from single pictures
        Latent Lerp (mtb): Linear interpolation (blend) between two latent vectors
        Image Remove Background Rembg (mtb): Removes the background from the input using Rembg.
        Qr Code (mtb): Basic QR Code generator
        Unsplash Image (mtb): Unsplash Image given a keyword and a size
        Text To Image (mtb): Utils to convert text to image using a font
        String Replace (mtb): Basic string replacement
        Fit Number (mtb): Fit the input float using a source and target range
        Get Batch From History (mtb): Very experimental node to load images from the history of the server.
        Any To String (mtb): Tries to take any input and convert it to a string
        Save Gif (mtb): Save the images from the batch as a GIF
        Export With Ffmpeg (mtb): Export with FFmpeg (Experimental)
        Transform Image (mtb): Save torch tensors (image, mask or latent) to disk, useful to debug things outside comfy
        Bbox From Mask (mtb): From a mask extract the bounding box
        Bbox (mtb): The bounding box (BBOX) custom type used by other nodes
        Crop (mtb): Crops an image and an optional mask to a given bounding box
        Uncrop (mtb): Uncrops an image to a given bounding box
[Allor]: 11 standard nodes was overridden.
[Allor]: 12 modules enabled.
[Allor]: 95 nodes was loaded.
### Loading: ComfyUI-Manager (V0.22.4)
### ComfyUI Revision: 1340 [30eb92c3]
### ComfyUI-Manager: Copy .js from '/home/ubuntu/ComfyUI/custom_nodes/ComfyUI-Manager/js/comfyui-manager.js' to '/home/ubuntu/ComfyUI/web/extensions/comfyui-manager'
### Loading: ComfyUI-Impact-Pack (V3.18)
### Loading: ComfyUI-Impact-Pack (Subpack: V0.2)
Total VRAM 22516 MB, total RAM 31751 MB
xformers version: 0.0.20
Set vram state to: NORMAL_VRAM
Device: cuda:0 NVIDIA A10G : cudaMallocAsync
WAS Node Suite: BlenderNeko's Advanced CLIP Text Encode found, attempting to enable `CLIPTextEncode` support.
WAS Node Suite: `CLIPTextEncode (BlenderNeko Advanced + NSP)` node enabled under `WAS Suite/Conditioning` menu.
WAS Node Suite: OpenCV Python FFMPEG support is enabled
WAS Node Suite Warning: `ffmpeg_bin_path` is not set in `/home/ubuntu/ComfyUI/custom_nodes/was-node-suite-comfyui/was_suite_config.json` config file. Will attempt to use system ffmpeg binaries if available.
WAS Node Suite: Finished. Loaded 192 nodes successfully.

        "Don't be afraid to give up the good to go for the great." - John D. Rockefeller

[tinyterraNodes] Loaded
Davemane42 Custom Nodes: Loaded
Using xformers cross attention

Import times for custom nodes:
   0.0 seconds: /home/ubuntu/ComfyUI/custom_nodes/rembg-comfyui-node
   0.0 seconds: /home/ubuntu/ComfyUI/custom_nodes/ComfyUI_Noise
   0.0 seconds: /home/ubuntu/ComfyUI/custom_nodes/SD-Latent-Upscaler
   0.0 seconds: /home/ubuntu/ComfyUI/custom_nodes/comfy_clip_blip_node
   0.0 seconds: /home/ubuntu/ComfyUI/custom_nodes/ComfyUI_TiledKSampler
   0.0 seconds: /home/ubuntu/ComfyUI/custom_nodes/ComfyUI_Cutoff
   0.0 seconds: /home/ubuntu/ComfyUI/custom_nodes/ComfyUI_ADV_CLIP_emb
   0.0 seconds: /home/ubuntu/ComfyUI/custom_nodes/ComfyI2I
   0.0 seconds: /home/ubuntu/ComfyUI/custom_nodes/masquerade-nodes-comfyui
   0.0 seconds: /home/ubuntu/ComfyUI/custom_nodes/ComfyUI_NestedNodeBuilder
   0.0 seconds: /home/ubuntu/ComfyUI/custom_nodes/ComfyUI_SeeCoder
   0.0 seconds: /home/ubuntu/ComfyUI/custom_nodes/ComfyMath
   0.0 seconds: /home/ubuntu/ComfyUI/custom_nodes/Derfuu_ComfyUI_ModdedNodes
   0.0 seconds: /home/ubuntu/ComfyUI/custom_nodes/clipseg.py
   0.0 seconds: /home/ubuntu/ComfyUI/custom_nodes/ComfyUI-Allor
   0.0 seconds: /home/ubuntu/ComfyUI/custom_nodes/ComfyUI_Dave_CustomNode
   0.0 seconds: /home/ubuntu/ComfyUI/custom_nodes/ComfyUI-Custom-Scripts
   0.0 seconds: /home/ubuntu/ComfyUI/custom_nodes/trNodes
   0.0 seconds: /home/ubuntu/ComfyUI/custom_nodes/ComfyUI_tinyterraNodes
   0.0 seconds: /home/ubuntu/ComfyUI/custom_nodes/comfyui-dynamicprompts
   0.1 seconds: /home/ubuntu/ComfyUI/custom_nodes/ComfyUI-Manager
   0.1 seconds: /home/ubuntu/ComfyUI/custom_nodes/ComfyUI_smZNodes
   0.2 seconds: /home/ubuntu/ComfyUI/custom_nodes/comfy_controlnet_preprocessors
   0.2 seconds: /home/ubuntu/ComfyUI/custom_nodes/text-test
   0.4 seconds: /home/ubuntu/ComfyUI/custom_nodes/efficiency-nodes-comfyui
   0.4 seconds: /home/ubuntu/ComfyUI/custom_nodes/ComfyUI_Inpaint
   0.8 seconds: /home/ubuntu/ComfyUI/custom_nodes/was-node-suite-comfyui
   1.1 seconds: /home/ubuntu/ComfyUI/custom_nodes/comfy_mtb
   1.4 seconds: /home/ubuntu/ComfyUI/custom_nodes/ComfyUI-Impact-Pack

Starting server

To see the GUI go to: http://0.0.0.0:8188
FETCH DATA from: /home/ubuntu/ComfyUI/custom_nodes/ComfyUI-Manager/extension-node-map.json
```

**Comment by ssitu** [comment]
I'm not aware of any conflicts. I would disable all the nodes that you aren't using in the workflow and delete everything in `ComfyUI\web\extensions` except for `core`, `comfyui-manager`, and `logging.js.example` and see if it still does that.

It might be useful if you open the developer console with F12 on the ComfyUI webpage and see what error pops up when you try to unnest the node.

**Comment by ssitu** [comment]
Nevermind, looks like I have the same issue. I'll get that fixed ASAP.

**Comment by ssitu** [comment]
This should be fixed with a723e1e4290822266111ffb4fd9bcddf45894458 and 09cef3787d3459c4fdfc755c3d9284ea47c41aef. The PR was working until I made changes which had a bug that converted hidden inputs to actual inputs in the nested node, messing up the inputs.
